### PR TITLE
feat: support list in non-nested mutable json

### PR DESCRIPTION
**Pull request by iloveitaly**

### CODE REVIEW

**Comment by edelooff** [comment]
This makes very explicit what essentially was an unintended (but perhaps _fine_) behavior that's currently in the library: the top-level isn't always a `dict`.

That was (is?) the basic assumption, but the way `NestedMutable` is implemented (to support easy nesting), the top-level for *that one* was allowed to be a list rather than a dict, creating a weird difference between the nested mutable and 'flat mutable'.

The databases seem to be fine with JSON arrays as the top level, so I'm not sure this library should arbitrarily stick to just objects. And even [RFC 4627](https://datatracker.ietf.org/doc/html/rfc4627) supports both arrays and objects at the top level, so this seems in line with that and should be supported.

File: sqlalchemy_json/__init__.py [hunk 1]
```diff
@@ -50,14 +50,28 @@ def coerce(cls, key, value):
             return NestedMutableList.coerce(key, value)
         return super(cls).coerce(key, value)
 
+class MutableListOrDict(Mutable):
```
Comment (edelooff) [inline sqlalchemy_json/__init__.py:53]: This should maintain 2 empty lines between definitions (and I guess I should really include `black` or `flake8` with a plugin in the CI runs..)

File: sqlalchemy_json/__init__.py [hunk 2]
```diff
@@ -50,14 +50,28 @@ def coerce(cls, key, value):
...
+        return super(cls).coerce(key, value)
 
 def mutable_json_type(dbtype=JSON, nested=False):
     """Type creator for (optionally nested) mutable JSON column types.
 
     The default backend data type is sqlalchemy.types.JSON, but can be set to
     any other type by providing the `dbtype` parameter.
     """
-    mutable_type = NestedMutable if nested else MutableDict
+    mutable_type = NestedMutable if nested else MutableListOrDict
```
Comment (edelooff) [inline sqlalchemy_json/__init__.py:74]: I'm not 100% sure on the name here, but that might be a bit of a wider concern, naming this `NestedJson` isn't possible as the name gets taken by the _column_ type later (which I think is more generally useful in exports), but getting it to better capture its meaning *and* then align it with the `Nested...` variant would be nice.

Suggestions are welcome if you have any.

File: test/test_sqlalchemy_json.py [hunk 3]
```diff
@@ -67,6 +67,20 @@ def author(session):
...
+        name="John Doe",
+        handles=["@JohnDoe", "JohnDoe"],
+    )
+    session.add(author)
+    session.commit()
+    session.refresh(author)
+
+    assert author.name == "John Doe"
+    assert author.handles == ["@JohnDoe", "JohnDoe"]
+
```
Comment (edelooff) [inline test/test_sqlalchemy_json.py:82]:
```suggestion
```

File: test/test_sqlalchemy_json.py [hunk 4]
```diff
@@ -156,3 +170,9 @@ def test_dict_merging(session, article):
             "someone/somerepo": 10,
         },
     }
+
+def test_mutable_json_list(session, author_with_list):
+    author_with_list.handles.append("@mike_bianco")
+    session.commit()
+
+    assert author_with_list.handles == ["@JohnDoe", "JohnDoe", "@mike_bianco"]
```
Comment (edelooff) [inline test/test_sqlalchemy_json.py:178]:
```suggestion
    assert author_with_list.handles == ["@JohnDoe", "JohnDoe", "@mike_bianco"]
    
```

File: sqlalchemy_json/__init__.py (hunk 1 above)
Comment (iloveitaly) [inline sqlalchemy_json/__init__.py:53]:
```suggestion

class MutableListOrDict(Mutable):
```

**Comment by iloveitaly** [comment]
> creating a weird difference between the nested mutable and 'flat mutable'.

yeah, this is what was unintuitive to me and the driver behind the change here.

Let me know if any other changes are needed!

**Review by edelooff — APPROVED** [review]

**Pull request merged**